File: haskell/problem9.hs
-- problem9.hs

-- Implementa una funcion que toma la eleccion del usuario
-- (vegetariana o no) y el ingrediente seleccionado,
-- devolviendo una descripcion de la pizza y si es vegetariana.

base :: [String]
base = ["mozzarella", "tomate"]

ingredientesVeget :: [String]
ingredientesVeget = ["Maiz tierno", "Platano maduro caramelizado", "Aguacate en laminas", "Queso costeno rallado"]

ingredientesNoVeget :: [String]
ingredientesNoVeget = ["Chorizo santarrosano", "Chicharron crocante", "Carne desmechada", "Pollo apanado con hogao"]

descripcionPizza :: Bool -> Int -> String
descripcionPizza esVeget opcion
    | opcion < 1 || opcion > length menu = "Error: opcion invalida"
    | otherwise = "Pizza " ++ tipo ++ " con " ++ ingredientesFinales
  where
    menu = if esVeget then ingredientesVeget else ingredientesNoVeget
    extra = menu !! (opcion - 1)
    tipo = if esVeget then "vegetariana" else "no vegetariana"
    ingredientesFinales = unwords (init (base ++ [extra])) ++ " y " ++ last (base ++ [extra])

main :: IO ()
main = do
    putStrLn "Desea una pizza vegetariana? (s/n): "
    r <- getLine
    let esVeget = map toLower r == "s"
    putStrLn "Seleccione un ingrediente adicional:"
    let menu = if esVeget then ingredientesVeget else ingredientesNoVeget
    mapM_ (\(i,ing) -> putStrLn (show i ++ ". " ++ ing)) (zip [1..] menu)
    opcionStr <- getLine
    let opcion = read opcionStr :: Int
    putStrLn (descripcionPizza esVeget opcion)
  where
    toLower c
        | c >= 'A' && c <= 'Z' = toEnum (fromEnum c + 32)
        | otherwise = c

File: python/problem9.py
# problem9.py
BASE = ["mozzarella", "tomate"]
ING_VEGET = ["Maiz tierno", "Platano maduro caramelizado", "Aguacate en laminas", "Queso costeno rallado"]
ING_NO_VEGET = ["Chorizo santarrosano", "Chicharron crocante", "Carne desmechada", "Pollo apanado con hogao"]

def descripcion_pizza(es_vegetariana: bool, opcion: int) -> str:
    """
    Devuelve la descripcion final de la pizza segun el tipo y el ingrediente elegido.
    """
    menu = ING_VEGET if es_vegetariana else ING_NO_VEGET
    if opcion < 1 or opcion > len(menu):
        return "Error: opcion de ingrediente invalida."
    
    extra = menu[opcion - 1]
    tipo = "vegetariana" if es_vegetariana else "no vegetariana"
    ingredientes = BASE + [extra]
    return f"Pizza {tipo} con " + ", ".join(ingredientes)


if __name__ == "__main__":
    try:
        respuesta = input("Desea una pizza vegetariana? (s/n): ").strip().lower()
        
        if respuesta not in ("s", "n"):
            print("Error: respuesta invalida, use 's' o 'n'.")
        else:
            es_vegetariana = respuesta == "s"
            menu = ING_VEGET if es_vegetariana else ING_NO_VEGET
            
            print("Elige un ingrediente adicional:")
            for i, ing in enumerate(menu, start=1):
                print(f"{i}. {ing}")
            
            opcion = int(input("Numero de ingrediente: "))
            print(descripcion_pizza(es_vegetariana, opcion))
    except ValueError:
        print("Error: debes ingresar un numero valido para el ingrediente.")
